hashednode.go

```diff
@@ -32,6 +32,7 @@ import (
 
 type HashedNode struct {
 	commitment []byte
+	resolution *Point
```

Comment:
I would prefer to call it `cachedPoint` or something that explicitly states it's a point